### ISSUE TITLE
Collisions with invisible vehicles

### ISSUE THREAD

**Issue by cicciosab97** [carla-simulator/carla#4779]
CARLA version: CARLA 0.9.8

Platform/OS: Ubuntu 18.04.3 LTS

Python: 3.7

Problem you have experienced: I have experienced problems using Driving Benchmark, Cexp, and Scenario Runner. I believe that there may be problems related to the simulator. In practice, it happens that during the Benchmark, collisions with non-visible vehicles occur. When I start NoCrash Benchmark, there are vehicle collisions that the RGB-camera cannot detect. The vehicles with which the collision occurs appear to be those that were previously in the collision position.
Could it be RGB camera rendering problem, or transmission of information from the simulator?

https://user-images.githubusercontent.com/58523560/138547469-ee29b262-5dc9-4fcf-9d2d-db3a1aecf591.mp4

**Comment by glopezdiest** [comment]
Hey @cicciosab97, could you explain in more detail what exact commands are you running? Also, I don't understand the aim of the video, as it is it a 6 second normal drive through the lane.

**Comment by cicciosab97** [comment]
Yes. @glopezdiest this is NoCrash Benchmark from [https://github.com/carla-simulator/driving-benchmarks/tree/carla_09_cexp](url), but I used it in combination with CARLA 0.9.8. I also used the Scenario Runner package found from the ghithub repository. It is a matter of 6 seconds of driving along the lane, but here the episode ends by collision. The video ends at the collision. I think this is a sensor rendering issue. Effectively analyzing the non-player agents through the measurements of the scene, a vehicle is actually present at the collision point, but it is not visible in the RGB and Depth cameras. This is not an isolated case, because all the episodes end for the same reason. Therefore the Benchmark is falsified.
